tests/unit/sdk/test_protocols_generator.py

```diff
@@ -0,0 +1,58 @@
+import pytest
+
+from infrahub_sdk import InfrahubClient
+from infrahub_sdk.protocols_generator.generator import CodeGenerator
+
+
+@pytest.mark.parametrize(
+    "sync,input,output",
+    [
+        (True, "CoreNode", "CoreNodeSync"),
+        (
+            True,
+            ["LineageSource", "CoreNode", "CoreObjectTemplate"],
+            ["LineageSource", "CoreNodeSync", "CoreObjectTemplate"],
+        ),
+        (False, "CoreNode", "CoreNode"),
+        (
+            False,
+            ["LineageSource", "CoreNode", "CoreObjectTemplate"],
+            ["LineageSource", "CoreNode", "CoreObjectTemplate"],
+        ),
+    ],
+)
+async def test_filter_syncify(sync, input, output):
+    assert CodeGenerator._jinja2_filter_syncify(value=input, sync=sync) == output
+    assert CodeGenerator._jinja2_filter_syncify(value=input, sync=sync) == output
```

Comment:
I liked this pattern of setting up dataclasses for the test cases, I think it adds clarity with regards to what the expected input fields are along with the possibility to give each test a relevant name.

https://github.com/opsmill/infrahub-sdk-python/blob/v1.10.2/tests/unit/sdk/test_template.py#L45-L81

Reply:
Good point I've updated the test